src/components/RangeDateField/hooks/useRangeDateFieldState.ts

```diff
@@ -0,0 +1,306 @@
+/* eslint-disable complexity */
+import React from 'react';
+
+import {isValid} from '@gravity-ui/date-utils';
+import type {DateTime} from '@gravity-ui/date-utils';
+
+import {
+    type BaseDateFieldState,
+    useBaseDateFieldState,
+} from '../../DateField/hooks/useBaseDateFieldState';
```

Comment:
I think it's bad practies to import something from internal component files.
It's better to add hook to component API (index file)
```
import {
    type BaseDateFieldState,
    useBaseDateFieldState,
} from '../../DateField';
```